File: packages/typia/src/transformers/features/http/CreateHttpAssertFormDataTransformer.ts
import { HttpAssertFormDataProgrammer } from "../../../programmers/http/HttpAssertFormDataProgrammer";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace CreateHttpAssertFormDataTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "http.createAssertFormData",
      write: HttpAssertFormDataProgrammer.write,
    });
}

File: packages/typia/native/transform/features/http/CreateHttpAssertFormDataTransformer.go
package http

import (
  shimast "github.com/microsoft/typescript-go/shim/ast"
  nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
  nativehttpprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/http"
  nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type createHttpAssertFormDataTransformerNamespace struct{}

var CreateHttpAssertFormDataTransformer = createHttpAssertFormDataTransformerNamespace{}

func (createHttpAssertFormDataTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
  return nativeinternal.GenericTransformer.Factory(nativeinternal.GenericTransformer_IProps{
    ITransformProps: props,
    Method:          "http.createAssertFormData",
    Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
      return nativehttpprogrammers.HttpAssertFormDataProgrammer.Write(x)
    },
  })
}
